### PR TITLE
fix: Swapped allowed events behavior in drag-canvas.

### PR DESCRIPTION
**Pull request by NotHexadecimal**
Allows events listed in `ALLOW_EVENTS` to pass instead of blocking them.

### CODE REVIEW

File: packages/pc/src/behavior/drag-canvas.ts [hunk 1]
```diff
@@ -267,9 +267,9 @@ export default {
       return;
     }
     if (ALLOW_EVENTS.indexOf(code.toLowerCase()) > -1) {
-      self.keydown = true;
-    } else {
       self.keydown = false;
+    } else {
+      self.keydown = true;
```
Comment (Yanyan-Wang) [inline packages/pc/src/behavior/drag-canvas.ts:272]: What is the reason for this change? Maybe the `ALLOW_EVENTS` is not incorrect semantically. I think the `ALLOW_EVENTS` should be named as `PREVENT_EVENT` instead of this change.

**Comment by Yanyan-Wang** [comment]
Thanks for your PR!  Please checkout my comment ;)

File: packages/pc/src/behavior/drag-canvas.ts (hunk 1 above)
Comment (Yanyan-Wang) [inline packages/pc/src/behavior/drag-canvas.ts:272]: And if you wanna block the behavior by default and enable it while pressing a key, I think the behavior should provide another config to allow the mode, e.g. `assistKeys: string[]`. When a user config drag-canvas with `assistKeys: ['shift']`, only when the 'shift' is being pressed, the canvas can be dragged. To achieve the purpose, the `onKeyDown` should be something like: 
```
onKeyDown(e: KeyboardEvent) {
    const self = this as any;
    const code = e.key;
    if (!code) {
      return;
    }

   if (PREVENT_EVENT.includes(code.toLowerCase())) {
     this.preventKeyDown = true;
   } else {
     this.preventKeyDown = false;
   }

    if (this.get('assistKeys').includes(code.toLowerCase())) {
      this.assistKeyDown = true;
    } else {
      this.assistKeyDown = false;
    }
},
```

Which means the `this.keydown` is discarded, use a semantically variables `this.assistKeyDown` and `this.assistKeyDown` instead. And line123, line173, line205 should be
```
if (!this.enabled()) return;
```

Where the `enabled` is something like:

```
enabled() {
  if (!!this.get('assistKeys')) { // the assistKeys is configured
    if (this.assistKeyDown) return true;
    return false;
  } else {
    if (this.preventKeyDown) return false;
    return true;
  }
}
```

And `onKeyUp`: 

```
onKeyUp() {
    this.assistKeyDown = false;
    this.preventKeyDown = false;
    //......
  },
```

**Comment by Aarebecca** [comment]
The PR didn't work for too long, so it was closed. If necessary, turn it on again.
